### DISCUSSION TITLE
sudo pip3 install virtualenvwrapper failed

### DISCUSSION THREAD

**Discussion by tanzhilangnw** [pypa/setuptools#3287] [Q&A]
the Errror info：
Traceback (most recent call last):
  File "/usr/local/bin/pip3", line 7, in <module>
    from pip._internal.cli.main import main
  File "/home/xp/.local/lib/python3.5/site-packages/pip/_internal/cli/main.py", line 57
    sys.stderr.write(f"ERROR: {exc}")
                                   ^
SyntaxError: invalid syntax

**Comment by abravalheri** [comment]
Hi @tanzhilangnw, I cannot replicate this error with:

```bash
> docker run --rm -it ubuntu:22.04

apt-get update
apt-get install -y python3 python3-pip
pip3 install virtualenvwrapper
```
This installs correctly.

Do you have a reproducer that we can investigate?

**Reply by tanzhilangnw** [reply]
Thanks for help! I have solved the problem by uninstall virtualenvwrapper and install virtualenvwrapper.

**Comment by abravalheri** [comment]
I also see that you are using `/home/xp/.local/lib/python3.5`.

Python 3.5 is no longer supported by `setuptools` and most of the packages in the ecosystem.
This version of Python reached its end of life in 2020, so it is very likely to find errors when trying to use it.

Have you tried using a version of Python that is still supported, e.g. 3.7, 3.8, 3.9, etc?
